python/cudf/cudf/core/frame.py

```diff
@@ -179,7 +179,14 @@ def _from_data(cls, data: MutableMapping) -> Self:
         """
         Construct cls from a ColumnAccessor-like mapping.
         """
-        obj = cls.__new__(cls)
+        from cudf.core.index import Index
+
+        if cls is Index:
+            # Because Index.__new__ is overridden as a constructor,
+            # to copy pandas...
+            obj = object.__new__(cls)
+        else:
+            obj = cls.__new__(cls)
```

Comment:
Is calling `cls.__new__` problematic for subclasses of `Index`?
I'm wondering if this risks re-entering `Index.__new__`.

Reply:
Yeah definitely. `pandas.Index` unfortunately has the same issue so (un)fortunately `cudf.Index` is equivalently problematic as pandas now 😅